jwql/shared_tasks/run_pipeline.py

```diff
@@ -0,0 +1,296 @@
+            # If the information is missing completely, put in a dummy value
+            datamodel.meta.dither.dither_points = 1
+
+        # Switch to calling the pipeline rather than individual steps,
+        # and use the run() method so that we can set parameters
+        # progammatically.
+        model = Detector1Pipeline()
+
+        # Always true
+        if instrument == 'nircam':
+            model.refpix.odd_even_rows = False
+
+        # Default CR rejection threshold is too low
+        model.jump.rejection_threshold = 15
+
+        # Turn off IPC step until it is put in the right place
+        model.ipc.skip = True
+
+        model.jump.save_results = True
+        model.jump.output_dir = work_directory
+        model.jump.maximum_cores = max_cores
+        jump_output = uncal_file.replace('uncal', 'jump')
+
+        # Check to see if the jump version of the requested file is already
+        # present
+        run_jump = not os.path.isfile(jump_output)
+
+        if ramp_fit:
+            model.ramp_fit.save_results = True
+            model.ramp_fit.maximum_cores = max_cores
+            # model.save_results = True
+            model.output_dir = work_directory
+            # pipe_output = os.path.join(output_dir, input_file_only.replace('uncal', 'rate'))
+            pipe_output = uncal_file.replace('uncal', '0_ramp_fit')
+            run_slope = not os.path.isfile(pipe_output)
+            if save_fitopt:
+                model.ramp_fit.save_opt = True
+                fitopt_output = uncal_file.replace('uncal', 'fitopt')
+                run_fitopt = not os.path.isfile(fitopt_output)
+            else:
+                model.ramp_fit.save_opt = False
+                fitopt_output = None
+                run_fitopt = False
+        else:
+            model.ramp_fit.skip = True
+            pipe_output = None
+            fitopt_output = None
+            run_slope = False
+            run_fitopt = False
+
+        if run_jump or (ramp_fit and run_slope) or (save_fitopt and run_fitopt):
+            model.run(datamodel)
+        else:
+            print(("Files with all requested calibration states for {} already present in "
+                   "output directory. Skipping pipeline call.".format(uncal_file)))
+    except Exception as e:
+        with open(status_file, "a+") as status_f:
+            status_f.write("EXCEPTION\n")
+            status_f.write("{}\n".format(e))
+            status_f.write("FAILED")
+        sys.exit(1)
+    
+    with open(status_file, "a+") as status_f:
+        status_f.write("{}\n".format(jump_output))
+        status_f.write("{}\n".format(pipe_output))
+        status_f.write("{}\n".format(pipe_output.replace("0_ramp", "1_ramp")))
+        status_f.write("{}\n".format(fitopt_output))
+        status_f.write("SUCCEEDED")
+    # Done.
+
+
+if __name__ == '__main__':
+    with open("/internal/data1/outputs/ops/calibrated_data/general_status.txt", "w") as status_file:
```

Comment:
@york-stsci is there anyway we can obscure this directory with a path from he config? Policy wise I don't think we are supposed to hard code paths/information about our servers in our code.

Reply:
Done.